### PR TITLE
[WIP] Add simple Nav

### PR DESCRIPTION
**Pull request by LorbusChris**
This is a proposal for a simple (nested) app navigation. Feel free to give some feedback!

The structure is as following, reusable components are on the third level. 
```
|-- container
  |--navigation
  {children}:
  |-- app
     |--title
     |--helloworld
  |-- view
     |--title
     |--test-data
```
Route Transitions work fine as far as I can tell. However, the actions from the test-data component wont fire just yet. 

It's using react-bootstrap(&-router) but there's no CSS yet.

What do you guys think?

Any input is appreciated :)

### CODE REVIEW

File: source/client/index.js [hunk 1]
```diff
@@ -26,10 +27,13 @@ const store = createStoreWithMiddleware(reducer);
...
 
 ReactDOM.render(
-  <Provider store={store}>
+  <Provider store={ store }>
     <Router history={ browserHistory }>
-      <Route path="/" component={ createApp(React) } />
-      <Route path="/test-data" component={ createTestData(React) } />
+      <Route path='/' component={ createContainer(React) }>
+        <IndexRoute component={ createHome(React) } />
+        <Redirect from='/home' to='/' />
```
Comment (ericelliott) [inline source/client/index.js]: Why is this needed?

Comment (LorbusChris) [inline source/client/index.js]: Both is part of the Router API. Since the Container does not include anything but the Navbar, I put in the IndexRoute to also show the Home(App) Component on "/". So thats really handy for nested components.

The Redirect is certainly not necessary here, but its also useful fuctionality imo. For example it could later be used to redirect international URLs like "/vista-de-data" -> "/data-view" (without a real translation, though) or obsolete URLs to their new locations. 

I'd like to be able to use them later on with the easy-API as well so I put them in here in the first place. If you think thats not desireable or necessary I can leave it out for now.

Comment (LorbusChris) [inline source/client/index.js]: Actually the Container contains a Navbar + Child. That way, on the IndexRoute "/", it renders the Home child.

Comment (ericelliott) [inline source/client/index.js]: > I'd like to be able to use them later on with the easy-API as well so I put them in here in the first place. If you think thats not desireable or necessary I can leave it out for now.

It's not necessary here. It should work just fine with the easy API, since it just takes regular react-router routes. All the features of the react-router API should work fine.

**Comment by LorbusChris** [comment]
Closing this for now. Will revisit once I find time.

**Comment by ericelliott** [comment]
:+1:
